File: core/src/com/qtech/bubbleblaster/common/References.java
package com.qtech.bubbleblaster.common;

import com.qtech.bubbleblaster.BubbleBlaster;

import java.io.File;

public class References {
    // Dirs
    public static final File QBUBBLES_DIR = BubbleBlaster.getGameDir();
    public static final File LOGS_DIR = new File(QBUBBLES_DIR.getAbsolutePath(), "Logs");
    public static final File ADDONS_DIR = new File(QBUBBLES_DIR.getAbsolutePath(), "Addons");
    //    public static final String SAVES_DIR = Path.join(QBUBBLES_DIR, "saves");
    public static final File SAVES_DIR = new File(QBUBBLES_DIR, "Saves");

    // Files
    public static final File SETTINGS_FILE = new File(QBUBBLES_DIR, "Settings.json");
    public static final File CRASH_REPORTS = new File(QBUBBLES_DIR, "CrashReports");

    static {
        //noinspection ResultOfMethodCallIgnored
        LOGS_DIR.mkdirs();
    }

    // Version
    public static final Version VERSION = new Version(1, 0, 0, VersionType.ALPHA, 0);
}

File: src/main/kotlin/qtech/bubbles/common/References.kt
package qtech.bubbles.common

import qtech.bubbles.BubbleBlaster
import java.io.File
import java.nio.file.Path
import java.nio.file.Paths

object References {
    // Dirs
    val QBUBBLES_DIR: File = BubbleBlaster.gameDir
    val LOGS_DIR: File = File(QBUBBLES_DIR.absolutePath, "Logs")
    val ADDONS_DIR: File = File(QBUBBLES_DIR.absolutePath, "Mods")

    val SAVES_DIR_PATH: Path = Paths.get(QBUBBLES_DIR.toString(), "saves");
    val SAVES_DIR: File = File(QBUBBLES_DIR, "Saves")

    // Files
    val SETTINGS_FILE: File = File(QBUBBLES_DIR, "Settings.json")
    val CRASH_REPORTS: File = File(QBUBBLES_DIR, "CrashReports")

    // Version
    val VERSION: Version = Version(1, 0, 0, VersionType.ALPHA, 0)

    init {
        LOGS_DIR.mkdirs()
    }
}